pkg/apis/batch/interfaces.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batch
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+func (obj *Job) PodSpec() (*api.PodSpec, string) {
```

Comment:
I would think external types should have this equivalence logic as well.

On Wed, Aug 24, 2016 at 2:15 PM, David Eads <notifications@github.com>
wrote:

> In pkg/apis/batch/interfaces.go
> <https://github.com/kubernetes/kubernetes/pull/31342#discussion_r76109882>
> :
>
> > +    http://www.apache.org/licenses/LICENSE-2.0
> > +
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package batch
> > +
> > +import (
> > +	"k8s.io/kubernetes/pkg/api"
> > +)
> > +
> > +func (obj *Job) PodSpec() (*api.PodSpec, string) {
>
> I vote against doing it this way. I would like to get rid of the internal
> types and this makes it harder.
>
> Instead why not just centralize the big switch statement somewhere?
>
> To do that and allow an API group to provide its own PodSpecer types
> would require plugin-style registration. Would I get to have that?
>
> I started this on the internal types (happens to be what we use today),
> but nothing precludes having this on the external type.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31342/files/505d0ab7915212e367601a51d580a98a5445de0e#r76109882>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p3YI8gh16qEkHZb97u5HJOcCD1Raks5qjIosgaJpZM4Jr9on>
> .
>